crates/matrix-sdk/src/room/timeline/mod.rs

```diff
@@ -125,13 +126,46 @@ impl Timeline {
         });
         let _fully_read_handler_guard = room.client.event_handler_drop_guard(fully_read_handle);
 
+        // Not using room.add_event_handler here because RoomKey events are
+        // to-device events that are not received in the context of a room.
+        let room_id = room.room_id().to_owned();
+        let room_key_handle = room.client.add_event_handler({
```

Comment:
One thing to note here, this is not the only source of room keys. As your test shows there's at least one other source.

In total we have three sources:
1. Receive them over to-device
2. Manually import them from a file
3. Import them from a backup

Your tests is using no. 2, no. 3 is another variant of 2 but not yet implemented.

I'm fine with opening an issue for this and linking to it, though adding some way for the `RoomKeyImportResult` to trigger a `retry_event_decryption()` call shouldn't be terribly complex either.